### PR TITLE
Fix enter key behavios on async forms

### PR DESCRIPTION
**Pull request by koertho**

### CODE REVIEW

File: src/Resources/npm-package/js/contao-filter-bundle.js [hunk 1]
```diff
@@ -61,7 +61,28 @@ class FilterBundle {
...
+
+    /**
+     * Register events for form async submit.
+     *
+     * These events need to be reintialized after form reload.
+     */
+    static registerFormAsyncSubmitEvents()
+    {
+        document.querySelector('.mod_filter form[data-async] input[type="text"]').addEventListener('keydown', (event) => {
+            if (event.code == 'Enter') {
```
Comment (AlexejKossmann) [inline src/Resources/npm-package/js/contao-filter-bundle.js:80]: can we be more strict here? or even use `event.which` to obtain numerical keycode?

Comment (koertho) [reply src/Resources/npm-package/js/contao-filter-bundle.js:80]: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent says event.which is deprecated. It is recommended to use event.key.

Comment (koertho) [reply src/Resources/npm-package/js/contao-filter-bundle.js]: Changed is to strict comparison with event.key. I think that's a good solution?

Comment (AlexejKossmann) [reply src/Resources/npm-package/js/contao-filter-bundle.js]: Ok, then is event.key the best solution, looks good for me

**Comment by koertho** [comment]
@Defcon0 Are you ok with this PR?

**Comment by Defcon0** [comment]
> @Defcon0 Are you ok with this PR?

Looks reasonable to me. If you guys tested it well, then I'm fine with it.

**Pull request merged**